src/components/CourseCard/CourseCard.tsx

```diff
@@ -0,0 +1,178 @@
+import classNames from 'classnames';
+import Hls from 'hls.js';
+import {
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Link } from 'react-router-dom';
+
+import { useAppDispatch } from '../../app/hooks';
+import { selectedCourseInit } from '../../features/selectedCourseSlice';
+import { Courses } from '../../types/Courses';
+import { RatingStars } from '../RatingStars';
+import './CourseCard.scss';
+
+type Props = {
```

Comment:
My little advice to keep types/interfaces/models in separate files (named as "models" for example).